lib/api/core.dart

```diff
@@ -53,14 +53,8 @@ class LiveApiConnection extends ApiConnection {
 
   @override
   Future<String> get(String route, Map<String, dynamic>? params) async {
-    final baseUrl = Uri.parse(auth.realmUrl);
-    final url = Uri(
-        scheme: baseUrl.scheme,
-        userInfo: baseUrl.userInfo,
-        host: baseUrl.host,
-        port: baseUrl.port,
-        path: "/api/v1/$route",
-        queryParameters: encodeParameters(params));
+    final url = auth.realmUrl.replace(
+        path: "/api/v1/$route", queryParameters: encodeParameters(params));
```

Comment:
This version is implicitly assuming that `realmUrl.fragment` is empty.

Similarly the code in `post` and in the two other routes is assuming `query` and `fragment` are both empty. (In those, the old code was also assuming that, though differently so.)

That assumption is fine, but let's make it explicit with an assert. Can put it on the Auth constructor:
```dart
  Auth({required this.realmUrl, required this.email, required this.apiKey})
    : assert(realmUrl.query.isEmpty && realmUrl.fragment.isEmpty);
```